pages/index.tsx

```diff
@@ -1,31 +1,123 @@
+import { useState, useEffect, useMemo, useCallback } from 'react';
+import Modal from 'react-modal';
 import styles from '../styles/Home.module.scss';
-import { home_page } from '../config';
+import { home_page, website_config } from '../config';
 // Internal Components
+import Image from '../components/Image';
 import NavBar from '../components/NavBar';
 import EventBanner from '../components/EventBanner';
 import HeroBanner from '../components/HeroBanner';
 import Footer from '../components/Footer';
 import Section from '../layouts/Section';
+import Link from 'next/link';
+
+Modal.setAppElement('#__next'); // Set the app element for accessibility
 
 export default function Home() {
   const { sections } = home_page;
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+
+  useEffect(() => {
```

Comment:
can you try using [useLayoutEffect](https://react.dev/reference/react/useLayoutEffect) but pay attention when trying this as it blocks dom pages (maybe just quickly run lighthouse and make sure nothing fails majorely when making this change). If you go witht he change I suggest above we can ignore this but I think this is the better solution.